src/spaced/spaced/game/weapon.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include <bave/logger.hpp>
+#include <bave/platform.hpp>
+#include <spaced/game/weapon_round.hpp>
+#include <spaced/services/layout.hpp>
+#include <spaced/services/services.hpp>
+
+namespace spaced {
+class Weapon : public bave::Polymorphic {
+  public:
+	using Round = IWeaponRound;
+
+	explicit Weapon(Services const& services, std::string name) : m_log{std::move(name)}, m_layout(&services.get<ILayout>()) {}
+
+	virtual auto fire(glm::vec2 muzzle_position) -> std::unique_ptr<Round> = 0;
+	[[nodiscard]] virtual auto is_idle() const -> bool = 0;
+
+	virtual void tick(bave::Seconds dt) = 0;
+
+	void inspect() {
+		if constexpr (bave::debug_v) { do_inspect(); }
+	}
+
+  protected:
+	[[nodiscard]] auto get_layout() const -> ILayout const& { return *m_layout; }
+	virtual void do_inspect() {}
+
+	bave::Logger m_log{};
+
+  private:
+	ILayout const* m_layout{};
```

Comment:
is this where all the stats (speed, damage, etc) will be stored?

Reply:
Not really, this is for storing the layout of the screen / world, for positioning things relative to the corners / edges, etc.